### PR TITLE
Add Functional February badge

### PR DESCRIPTION
**Pull request by iHiD**

### CODE REVIEW

File: app/models/badges/functional_february_badge.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    def self.worth_queuing?(exercise:)
+      TRACK_SLUGS.include?(exercise.track.slug)
+    end
+
+    def award_to?(user)
+      user.solutions.completed.joins(exercise: :track).
+        where('tracks.slug': TRACK_SLUGS).
+        where('MONTH(completed_at) = 2').
+        count >= 5
+    end
```
Comment (ErikSchierboom) [inline app/models/badges/functional_february_badge.rb]: Just as a check: we don't only award this badge to people that are participating in the challenge?

File: test/models/badges/functional_february_badge_test.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
+require "test_helper"
+
+class Badges::FunctionalFebruaryBadgeTest < ActiveSupport::TestCase
+  test "attributes" do
+    badge = create :functional_february_badge
+    assert_equal "Functional February", badge.name
+    assert_equal :ultimate, badge.rarity
```
Comment (ErikSchierboom) [inline test/models/badges/functional_february_badge_test.rb:7]: It this `ultimate` or `rare`? It's hard to determine this beforehand I feel.

**Review by ErikSchierboom — APPROVED** [review]

File: test/models/badges/functional_february_badge_test.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    assert_equal :functional, badge.icon
+    assert_equal 'Completed five exercises in a functional language in February', badge.description
```
Comment (ErikSchierboom) [inline test/models/badges/functional_february_badge_test.rb]: I wonder if we should make it clear that it is only for functional languages featured in the Functional February theme (e.g. LISPS are functional but don't count)

```suggestion
    assert_equal 'Completed five exercises in a functional language in February', badge.description
```

**Review by dem4ron — APPROVED** [review]

File: app/models/badges/functional_february_badge.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    seed "Functional February",
+      :ultimate,
+      'functional',
+      'Completed five exercises in a functional language in February'
+
+    def self.worth_queuing?(exercise:)
+      TRACK_SLUGS.include?(exercise.track.slug)
+    end
+
+    def award_to?(user)
```
Comment (ErikSchierboom) [inline app/models/badges/functional_february_badge.rb:16]: Where is the award being awarded?

File: app/models/badges/functional_february_badge.rb (hunk 1 above)
Comment (iHiD) [reply app/models/badges/functional_february_badge.rb]: Correct. But thanks for checking!

File: test/models/badges/functional_february_badge_test.rb (hunk 2 above)
Comment (iHiD) [reply test/models/badges/functional_february_badge_test.rb:7]: Yeah, we can maybe change it later?

File: test/models/badges/functional_february_badge_test.rb (hunk 3 above)
Comment (iHiD) [reply test/models/badges/functional_february_badge_test.rb]: Yeah. We have no more characters spare though :S

File: test/factories/badges.rb [hunk 5]
```diff
@@ -11,6 +11,8 @@
     mentor researcher v3_pioneer tooling_pioneer
     moss begetter bard architect troubleshooter
     completer conceptual supermentor
+
```
Comment (ErikSchierboom) [inline test/factories/badges.rb]:
```suggestion
```

File: app/models/badges/functional_february_badge.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
+module Badges
+  class FunctionalFebruaryBadge < Badge
+    TRACK_SLUGS = %w[
+      clojure elixir erlang fsharp haskell ocaml scala sml gleam
+    ].freeze
+
+    seed "Functional February",
+      :ultimate,
+      'functional',
+      'Completed five exercises in a functional language in February'
```
Comment (ErikSchierboom) [inline app/models/badges/functional_february_badge.rb]:
```suggestion
      'Completed and published five exercises in a functional language in February'
```

File: test/models/badges/functional_february_badge_test.rb [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
```
Comment (ErikSchierboom) [inline test/models/badges/functional_february_badge_test.rb]:
```suggestion
    assert_equal 'Completed and published five exercises in a functional language in February', badge.description
```

**Pull request merged**